core/src/services/oss/core.rs

```diff
@@ -225,14 +229,17 @@ impl OssCore {
         }
 
         if let Some(cache_control) = args.cache_control() {
-            req = req.header(CACHE_CONTROL, cache_control)
+            req = req.header(CACHE_CONTROL, cache_control);
         }
 
-        // set sse headers
-        req = self.insert_sse_headers(req);
+        if let Some(user_metadata) = args.user_metadata() {
+            for (key, value) in user_metadata {
+                // before insert user defined metadata header, add prefix to the header name
+                req = req.header(format!("{}{}", constants::USER_METADATA_PREFIX, key), value)
```

Comment:
According to https://help.aliyun.com/zh/oss/developer-reference/putobject, the key of user defined metadata has some constrains, should we do some checks here ? otherwise the `aliyun` might return `SignatureDoesNotMatch` error, which can be confusing.

Reply:
> the key of user defined metadata has some constrains, should we do some checks here ?

That would be nice.